src/utils.ts

```diff
@@ -1,57 +1,52 @@
 import type { FieldAttribute } from "better-auth/db";
-import { StringRecordId } from "surrealdb";
+import type { Where } from "better-auth/types";
+
+export const operatorMap: Record<Required<Where>['operator'], string | null> = {
+    "eq": "==",
+    "ne": "!=",
+    "lt": "<",
+    "lte": "<=",
+    "gt": ">",
+    "gte": ">=",
+    "contains": "CONTAINS",
+    "in": "IN",
+    // not operators but functions
+    "starts_with": null,
+    "ends_with": null,
+}
 
-function isDateString(dateString: string) {
-	const date = new Date(dateString);
-	return !isNaN(date.getTime());
+export const typeMap: Record<string, string> = {
+    string: "string",
+    boolean: "bool",
+    number: "number",
+    date: "datetime",
+    "number[]": "array<number>",
+    "string[]": "array<string>",
 }
 
-function shouldConvertToRecordId(fieldName: string | undefined, model?: string): boolean {
-	if (!fieldName?.endsWith("Id")) {
-		return false;
-	}
-	
-	const excludedFields = ["providerId", "activeOrganizationId"];
-	if (excludedFields.includes(fieldName)) {
-		return false;
-	}
-	
-	// Special case: accountId should not be converted when model is "account"
-	if (fieldName === "accountId" && model === "account") {
-		return false;
-	}
-	
-	return true;
+function isDateString(dateString: string) {
+    const date = new Date(dateString);
+    return !isNaN(date.getTime());
 }
 
 export function withApplyDefault(
-	value: any,
-	field: FieldAttribute,
-	action: "create" | "update",
-	model?: string,
+    value: any,
+    field: FieldAttribute,
+    action: "create" | "update",
 ) {
-	switch (true) {
-		case action === "update":
-			return value;
-
-		case value === undefined || value === null:
-			if (field.defaultValue) {
-				return typeof field.defaultValue === "function"
-					? field.defaultValue()
-					: field.defaultValue;
-			}
-			return value;
-
-		case field.references?.model !== undefined:
-			return new StringRecordId(value);
-
-		case shouldConvertToRecordId(field.fieldName, model):
```

Comment:
this is necessary check otherwise its turns all the other frields to date? 
for example:  that generated  with the organization plugin. 
```
{
	createdAt: d'2025-08-15T11:03:18.098Z',
	id: organization:qX5rURLbbKYY0UKFr1oun0wNM1ZI8H4G,
	name: '2000-12-31T16:00:00Z',
	slug: '2000-12-31T16:00:00Z'
}
``` 

```surql
{
	createdAt: d'2025-08-15T11:03:18.098Z',
	default: true,
	id: team:5s4baf4rwgpuhj7lyh3u,
	isPrivate: false,
	metadata: {},
	name: '2000-12-31T16:00:00Z',
	organizationId: organization:qX5rURLbbKYY0UKFr1oun0wNM1ZI8H4G,
	slug: '2000-12-31T16:00:00Z'
}
```

Reply:
Just realized this is unused code, so it should be removed actually. The type conversions are all handled by the `createAdapter` function now.

Reply:
I did find the date-string issue though, will look to see what is going on shortly.

Reply:
Changed my approach, I utilize variables in the `convertWhereClause` and the `createAdapter` function handles conversions of fields as needed.